### ISSUE TITLE
Bug: (Windows) Parsing error: inferred tsconfigRootDir should be a resolved absolute path, but received: "C:/Users/29422/Desktop/vue-project".

### ISSUE THREAD

**Issue by lvzhenbo** [typescript-eslint/typescript-eslint#11530]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

https://github.com/typescript-eslint/typescript-eslint/pull/11463

### Reproduction Repository Link

https://github.com/lvzhenbo/vue-project

### Repro Steps

1) clone the repo
2) `pnpm install`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.41.0` |
| `@typescript-eslint/parser`            | `8.41.0` |
| `@typescript-eslint/scope-manager`     | `8.41.0` |
| `@typescript-eslint/typescript-estree` | `8.41.0` |
| `@typescript-eslint/type-utils`        | `8.41.0` |
| `@typescript-eslint/utils`             | `8.41.0` |
| `TypeScript`                           | `5.9.2` |
| `ESLint`                               | `9.34.0` |
| `node`                                 | `22.18.0` |

**Comment by kirkwaiblinger** [comment]
Grr, windows paths!! :rage:

Thanks for reporting!

Seems the issue here is to do with the stack trace API sometimes producing paths with `/` for pathsep on windows rather than `\`. Should be quite simple to normalize, but I'd like to understand which conditions trigger the `/` pathsep to be used (i.e. whether it's `file://` URLs or ordinary paths in the stack trace). 

https://github.com/typescript-eslint/typescript-eslint/blob/7319bad3a5022be2adfbcb331451cfd85d1d786a/packages/typescript-eslint/src/getTSConfigRootDirFromStack.ts#L29-L48

**Comment by lvzhenbo** [comment]
@kirkwaiblinger 
```
stackFrameFilePathOrUrl: file:///C:/Users/htk/Desktop/vue-project/test.ts
stackFrameFilePath: C:\Users\htk\Desktop\vue-project\test.ts
parsedPath: {
  "root": "C:\\",
  "dir": "C:\\Users\\htk\\Desktop\\vue-project",
  "base": "test.ts",
  "ext": ".ts",
  "name": "test"
}
stackFrameFilePathOrUrl: node:internal/modules/esm/module_job
stackFrameFilePath: node:internal/modules/esm/module_job
parsedPath: {
  "root": "",
  "dir": "node:internal/modules/esm",
  "base": "module_job",
  "ext": "",
  "name": "module_job"
}
stackFrameFilePathOrUrl: node:internal/modules/esm/loader
stackFrameFilePath: node:internal/modules/esm/loader
parsedPath: {
  "root": "",
  "dir": "node:internal/modules/esm",
  "base": "loader",
  "ext": "",
  "name": "loader"
}
stackFrameFilePathOrUrl: node:internal/modules/run_main
stackFrameFilePath: node:internal/modules/run_main
parsedPath: {
  "root": "",
  "dir": "node:internal/modules",
  "base": "run_main",
  "ext": "",
  "name": "run_main"
}
```

**Comment by kirkwaiblinger** [comment]
@lvzhenbo 

Oh, hmmmm, that... is exactly what I "expected". But I'd have thought in order to hit the assertion you're seeing it must be the case that `parsedPath.dir` is `"C:/a/b/c"` rather than `"C:\\a\\b\\c"`, since `parsedPath.dir` is what gets returned and printed 👀 

https://github.com/typescript-eslint/typescript-eslint/blob/270fa594102856ca9cc06ca897819c3c196c1314/packages/typescript-estree/src/parseSettings/createParseSettings.ts#L55-L65

---

_Unless_ - are you somehow hitting the branch where it prints `process.cwd()`, and that's written with `/` on windows?? (this shouldn't be the case, since you have a config file matching the pattern)

**Comment by lvzhenbo** [comment]
<img width="1559" height="1289" alt="Image" src="https://github.com/user-attachments/assets/8540ef5f-8d3d-439c-9d0c-0859e3344b1d" alt="editor" />

It's strange, it seems like everything should be like this, but the editor gives the opposite feedback

**Comment by lvzhenbo** [comment]
@kirkwaiblinger I found something wrong with eslint's vscode plugin

<img width="1593" height="716" alt="vscode eslint output" src="https://github.com/user-attachments/assets/bfd8cc4b-ff57-4412-b713-3ab536c85eef" />

**Comment by pennwang** [comment]
<img width="1135" height="365" alt="Image" src="https://github.com/user-attachments/assets/316f58cc-12cf-4d55-a4ef-85631a54f321" />

**Comment by kirkwaiblinger** [comment]
Oh, very curious! I didn't actually test out your reproduction yet since I don't have access to a windows machine for the moment (can look in a week or so when I'm no longer travelling). Sounds like this might need a bit of a deep dive.

**Comment by posti85** [comment]
I'm facing the same issue. Are there any workaround to apply until it is fixed?

**Comment by JoshuaKGoldberg** [comment]
As a workaround in the meantime, you should be able to explicitly set [`tsconfigRootDir`](https://typescript-eslint.io/packages/parser#tsconfigrootdir):

```ts
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  eslint.configs.recommended,
  tseslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
        tsconfigRootDir: import.meta.dirname, // 👈
      },
    },
  },
);
```

N.b. #11532 tracks filling out that specific docs entry. In the meantime, https://typescript-eslint.io/getting-started/typed-linting has more information.

**Comment by mayorbyrne** [comment]
> I'm facing the same issue. Are there any workaround to apply until it is fixed?

For what it's worth, you can also install the previous version in your project dev-dependencies. This seemed to work for me in the interim, though it might be more maintenance once the problem is fixed. `npm install typescript-eslint@8.40.0 --save-dev`

**Comment by posti85** [comment]
> For what it's worth, you can also install the previous version in your project dev-dependencies.

Yes, but typescript-eslint is a dependency of other dependency I already have. I don't want to  install it explicitly in my project.

**Comment by lsy934127847** [comment]
I encountered the same problem in the Vite project ， Also specified tsconfigRootDir can reslove

**Comment by lsy934127847** [comment]
> As a workaround in the meantime, you should be able to explicitly set [`tsconfigRootDir`](https://typescript-eslint.io/packages/parser#tsconfigrootdir):
> 
> import eslint from '@eslint/js';
> import tseslint from 'typescript-eslint';
> 
> export default tseslint.config(
>   eslint.configs.recommended,
>   tseslint.configs.recommendedTypeChecked,
>   {
>     languageOptions: {
>       parserOptions: {
>         projectService: true,
>         tsconfigRootDir: import.meta.dirname, // 👈
>       },
>     },
>   },
> );
> N.b. [#11532](https://github.com/typescript-eslint/typescript-eslint/issues/11532) tracks filling out that specific docs entry. In the meantime, https://typescript-eslint.io/getting-started/typed-linting has more information.

I encountered the same problem in the Vite project ， Also specified tsconfigRootDir can reslove

<img width="536" height="150" alt="Image" src="https://github.com/user-attachments/assets/85c9bcbd-13f0-4a4e-be18-f83c4041287e" />

**Comment by mhlsky** [comment]
这是来自QQ邮箱的自动回复邮件。已收----梦里飞Mhlsky(孟hl)

**Comment by lsy934127847** [comment]
I encountered the same problem in the Vite project ， Also specified tsconfigRootDir can reslove

<img width="536" height="150" alt="Image" src="https://github.com/user-attachments/assets/85c9bcbd-13f0-4a4e-be18-f83c4041287e" />

**Comment by Jason3S** [comment]
This error is not limited to Windows. I'm getting it on Ubuntu as well.


The suggestion https://github.com/typescript-eslint/typescript-eslint/issues/11530#issuecomment-3227928917 works for me.

**Comment by kirkwaiblinger** [comment]
Ok, @lvzhenbo, I've broken out the windows laptop and this appears to be triggered by the use of a TS config file via jiti. When a jiti TS file is in the call stack, it gives its path as `C:a/b/c`, (not `file:///C:a/b/c`, which we're equipped to handle, and not `C:\a\b\c`, which we'd also expect). 

I'm inclined to say that this is a bug in jiti, but we can also be more permissive on our side in order to prevent the assertion from firing.

**Comment by kirkwaiblinger** [comment]
Also of course using a mjs config, or otherwise avoiding jiti, is a valid workaround in the meantime

**Comment by lvzhenbo** [comment]
The use of jiti is expected to persist for some time until the built-in Node version in VS Code is updated to 22.18 or higher.

<img width="380" height="306" alt="vscode 103" src="https://github.com/user-attachments/assets/e5c7592b-5084-49ba-bcb2-f20763d36b7b" />

<img width="382" height="306" alt="vscode 104" src="https://github.com/user-attachments/assets/b9686956-3f3a-4751-9cca-074398cf2fe0" />

**Comment by buggedcom** [comment]
> As a workaround in the meantime, you should be able to explicitly set [`tsconfigRootDir`](https://typescript-eslint.io/packages/parser#tsconfigrootdir):
> 
> import eslint from '@eslint/js';
> import tseslint from 'typescript-eslint';
> 
> export default tseslint.config(
>   eslint.configs.recommended,
>   tseslint.configs.recommendedTypeChecked,
>   {
>     languageOptions: {
>       parserOptions: {
>         projectService: true,
>         tsconfigRootDir: import.meta.dirname, // 👈
>       },
>     },
>   },
> );
> N.b. [#11532](https://github.com/typescript-eslint/typescript-eslint/issues/11532) tracks filling out that specific docs entry. In the meantime, https://typescript-eslint.io/getting-started/typed-linting has more information.

I'm running windows on a Parallels VM on a mac, I don't think it matters but have noted just in case.

For me, `import.meta.dirname` still outputs paths with `/` in it and the code still produces the same errors.

For example

```
C:/Users/ollie/xxxx/apps/my-app
```
however passing it through `path.resolve(tsconfigRootDir)` converts the `/` to a correct windows path.

```
C:\Users\ollie\xxx\apps\my-app
```
and the errors stop.

```js
import eslint from '@eslint/js';
import tseslint from 'typescript-eslint';
import path from 'node:path';

export default tseslint.config(
  eslint.configs.recommended,
  tseslint.configs.recommendedTypeChecked,
  {
    languageOptions: {
      parserOptions: {
        projectService: true,
        tsconfigRootDir: path.resolve(import.meta.dirname), // 👈
      },
    },
  },
);
```

**Comment by kirkwaiblinger** [comment]
Hi all!

#11546 has just been merged which solves the reproduction in the OP. This [will be released in a few hours](https://typescript-eslint.io/users/releases). For anyone that continues to experience issues around this, I'd ask you to double check whether it's resolved in the new version, and, if not, please open a new issue with a reproduction, or chat with us on discord for help requests.

Thank you!
